### ISSUE TITLE
Velocities should be properly initialised to 0

### ISSUE THREAD

**Issue by christofferaakre** [aephil/WeLD#30]
This issue may be linked to this other issue: https://github.com/aephil/WeLD/issues/28

One weird thing I noticed straight away when trying to debug this is that the nodes seem to start with non-zero `vi`, when it should be zero according to the code in `modules/physics/Lattice.js`, right? Even if I do something like
```javascript
lattice.data.forEach(d => {
    d.vi.x = 0;
    d.vi.y = 0;
    d.vi.z = 0;
})
console.log(lattice.data);
``` 
right after your
```javascript
// TESTING: push the first node in the x direction to simulate an initial Extension
lattice.data[0].ri.x += 0.5;
```
I still see non-zero `vi` for both nodes. Any idea why this is the case?

_Originally posted by @christofferaakre in https://github.com/aephil/WeLD/issues/28#issuecomment-968768020_

**Comment by christofferaakre** [comment]
@TakudzwaMakoni It gets weirder: Try running this code right before you move the first node to the right by 0.5:
```javascript
lattice.data.forEach(d => {                                              
     d.vi.x = 0;                                                                 
     d.vi.y = 10;                                                          
     d.vi.z = 10;                                                          
   })                                                                       
                                                                               
   console.log(lattice.data[0]);
```
When I try this, I get a random value for the x part, and a number just slightly less than 10 for the y and z parts when inspecting `vi`. Now, log out the actual `vi` instead of the entire node object:
```javascript
lattice.data.forEach(d => {                                              
      d.vi.x = 0;                                                           
      d.vi.y = 10;                                                          
      d.vi.z = 10;                                                          
   })                                                                       
                                                                            
   console.log(lattice.data[0].vi);
```
With this,
 I get `{x: 0, y: 10, z: 10}` as expected every time!
I have no idea why these two would give a different result
![whole-object](https://user-images.githubusercontent.com/31142852/142198699-ee77dd08-a2d8-4ba6-8cb1-6a856901e61a.png)
![vi](https://user-images.githubusercontent.com/31142852/142198714-ecb53e6f-f055-423d-9704-d39f09634e2c.png)

**Comment by TakudzwaMakoni** [comment]
setting physics to empty array on line 131 seems to make this go away, even though the data is being logged before any rendering. Issue seems to be coming from `verletController.velocityVerlet` is my first guess

**Comment by TakudzwaMakoni** [comment]
It seems to be that since we are passing in a reference to the value, it is showing us the value after it is modified by a render call much later on line 155.

I am pretty sure this is what is happening, because when i check both values before and after velocityVerlet and updateState, the values are still there. only after update state is called, can vi ever actually change, so commenting that function out exposes that the values persist inside both data[0], and data[0].vi

here is some more information:

https://developer.mozilla.org/en-US/docs/Web/API/Console/log

and this quote:

> Please be warned that if you log objects in the latest versions of Chrome and Firefox what you get logged on the console is a reference to the object, which is not necessarily the 'value' of the object at the moment in time you call console.log(), but it is the value of the object at the moment you open the console.

in short by the time we are seeing lattice.data[0] in the console, renderer has already called the physics functions at least once, modifying the data

**Comment by TakudzwaMakoni** [comment]
after testing, this seems to be the case, when you are comfortable to do so, you can probably close this one

**Comment by christofferaakre** [comment]
I see. This makes sense. It also explains why we get different values when logging just the vi vs. logging the entire object. Good find! I'll close this issue now.
